spec/organizations_spec.rb

```diff
@@ -231,4 +233,88 @@
         set_to_check.length.should eq 3
     end
 
+    it "should not allow duplicated integration ids" do
+        # given
+        model =  MoveToGo::RootModel.new
+
+        organization = MoveToGo::Organization.new
+        organization.name = "name1"
+        organization.integration_id = "1337"
+        model.add_organization(organization)
+
+        organization = MoveToGo::Organization.new
+        organization.name = "name2"
+        organization.integration_id = "1337"
+
+        assert_raise(MoveToGo::AlreadyAddedError) {
+            model.add_organization(organization)
+        }
+    end
+
+    it "should allow different integration ids" do
+        # given
+        model =  MoveToGo::RootModel.new
+
+        organization = MoveToGo::Organization.new
+        organization.name = "name1"
+        organization.integration_id = "1337"
+        model.add_organization(organization)
+
+        organization = MoveToGo::Organization.new
+        organization.name = "name2"
+        organization.integration_id = "1338"
+
+        assert_nothing_raised(MoveToGo::AlreadyAddedError) {
+            model.add_organization(organization)
+        }
+    end
+
+    it "should not allow duplicated source ids" do
+        # given
+        model =  MoveToGo::RootModel.new
+
+        organization = MoveToGo::Organization.new
+        organization.name = "name1"
+        organization.integration_id = "1337"
+        organization.with_source do |source|
+            source.par_se('1111')
+        end
+        model.add_organization(organization)
+
+        organization = MoveToGo::Organization.new
+        organization.name = "name2"
+        organization.integration_id = "1338"
+        organization.with_source do |source|
+            source.par_se('1111')
+        end
+        model.add_organization(organization)
+
+        # when, then
+        model.organizations.find_duplicates_by("source.id").length.should be > 0
```

Comment:
Det är lite konstigt att testen ovan "should not allow duplicated integration ids" verifierar det genom att fånga felet som kastas när organizationen läggs till medan testet "should not allow duplicated source ids" verifierar att det finns flera företag med samma source.id.

Egentligen skulle det väl kastas ett fel om rootmodel.add_organization anropas med ett företag vars source.id redan finns i rootmodellen?